### PR TITLE
Iox #471 restructure scripts

### LINKED ISSUE #471: Evaluation and restructuring of tools
## Brief feature description
For making iceoryx more stable, we need to look for better tooling to do code analysis and support the developer.
We should also take a deeper look into the ROS side what there is used, e.g. ament_lint.
Proposals are welcome!

List of common tools from ROS2:
* `ament_cmake_copyright <https://github.com/ament/ament_lint/blob/master/ament_cmake_copyright/doc/index.rst>`_ : a copyright linter which checks that copyright statements and license headers are present and correct (already integrated)

* `ament_cmake_cppcheck <https://github.com/ament/ament_lint/blob/master/ament_cmake_cppcheck/doc/index.rst>`_ : a C++ checker which can also find some logic tests (Not to be used because there is already clang-tidy)

* `ament_cmake_cpplint <https://github.com/ament/ament_lint/blob/master/ament_cmake_cpplint/doc/index.rst>`_ : a C++ style checker (e.g. comment style)

* `ament_cmake_lint_cmake <https://github.com/ament/ament_lint/blob/master/ament_cmake_lint_cmake/doc/index.rst>`_ : a cmake linter (Not to be used with ament to avoid hard ROS dependency)

* `ament_cmake_uncrustify <https://github.com/ament/ament_lint/blob/master/ament_cmake_uncrustify/doc/index.rst>`_ : a C++ style checker (Not to be used because there is already clang-format)

Additional ToDos for tooling:

- [x] restructure scripts in tools folder
- [x] lint shell scripts
- [x] tidy-up CI jobs
- [x] update dockerfile to Ubuntu Focal LTS

### LINKED ISSUE #920: Provide clean-only parameter in build script
## Brief feature description

@neillangmead41 requested a `clean-only` option in  `./tools/iceoryx_build_test.sh`

## Detailed information

* Adapt `./tools/iceoryx_build_test.sh` accordingly

### PR DESCRIPTION
**Pull request by dkroenke**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CHANGELOG.md

## Notes for Reviewer

- restructure scripts in tools folder (moving bash scripts into tools/scripts)
- lint shell scripts
- tidy-up CI jobs
- update dockerfile to Ubuntu Focal LTS
- spell checking on some documents

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #471 
- Closes #920

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -191,25 +191,25 @@ Integration tests are composition of more than one class and test their interact
...
 sudo apt install lcov
 ```
 
-In iceoryx we have multiple testlevels for testcoverage: 'unit', 'integration', 'component' and ’all’ for all testlevels together. You can create reports for these different testlevels or for all tests. Coverage is done with gcc.
+In iceoryx we have multiple test levels for test coverage: 'unit', 'integration', 'component' and ’all’ for all test levels together. You can create reports for these different test levels or for all tests. Coverage is done with gcc.
 The coverage scan applies to Quality level 3 and partly level 2 with branch coverage.
 
 For having a coverage report iceoryx needs to be compiled with coverage flags and the tests needs to be executed.
-You can do this with one command in iceroyx folder like this:
+You can do this with one command in iceoryx folder like this:
```
Comment (elBoberido) [inline CONTRIBUTING.md:205]: oh no, iceroyx is no more

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -264,7 +261,7 @@ With a comment in the same line:
     *mynullptr = foo; // PRQA S 4242 # Short description why
 ```
 
-With a comment one line above (with the number after the warning number, next ’n’ lines are inclusive)
+With a comment one line above (with the number after the warning number,next ’n’ lines are inclusive)
```
Comment (elBoberido) [inline CONTRIBUTING.md:264]: Don't we want a white-space here?

File: tools/iceoryx_build_test.sh [hunk 3]
```diff
@@ -258,23 +258,23 @@ fi
 if [ $CLEAN_BUILD == true ]
 then
     echo " [i] Cleaning build directory"
-    cd $WORKSPACE
-    rm -rf $BUILD_DIR/*
+    cd "$WORKSPACE"
+    rm -rf "${BUILD_DIR:?}/"*
```
Comment (elBoberido) [inline tools/iceoryx_build_test.sh:262]: Does this check whether `BUILD_DIR` is empty?

File: tools/iceoryx_build_test.sh [hunk 4]
```diff
@@ -332,34 +331,35 @@ if [ "$OUT_OF_TREE_FLAG" == "ON" ]; then
...
 fi
 
 for COMPONENT in $COMPONENTS; do
 
     case "$1" in
-        "unit" | "all")
+        "unit")
             if [ ! -f testresults/"$COMPONENT"_ModuleTestResults.xml ]; then
-                echo "xml:"$COMPONENT"_ModuletestTestResults.xml not found!"
+                echo "xml:""$COMPONENT""_ModuletestTestResults.xml not found!"
```
Comment (elBoberido) [inline tools/iceoryx_build_test.sh:356]: Can this be simplified?
```suggestion
                echo "xml:${COMPONENT}_ModuletestTestResults.xml not found!"
```

File: tools/scripts/iceoryx_release.sh [hunk 5]
```diff
@@ -41,11 +41,11 @@ done
 target_version=$(head -n 1 "$WORKSPACE/VERSION")
 echo "The new version number will be: $target_version"
 
-find $WORKSPACE -name "*.xml" -type f -exec sed -i 's/<version>\(.*\)<\/version>/<version>'$target_version'<\/version>/g' {} \;
+find "$WORKSPACE" -name "*.xml" -type f -exec sed -i 's/<version>\(.*\)<\/version>/<version>'"$target_version"'<\/version>/g' {} \;
 
-find $WORKSPACE -name "CMakeLists.txt" -type f -exec sed -i 's/set(IOX_VERSION_STRING "\(.*\)")/set\(IOX_VERSION_STRING "'$target_version'"\)/g' {} \;
+find "$WORKSPACE" -name "CMakeLists.txt" -type f -exec sed -i 's/set(IOX_VERSION_STRING "\(.*\)")/set\(IOX_VERSION_STRING "'"$target_version"'"\)/g' {} \;
```
Comment (elBoberido) [inline tools/scripts/iceoryx_release.sh:46]: Can some of the quotes around `$target_version` be removed?

File: tools/iceoryx_build_test.sh (hunk 3 above)
Comment (dkroenke) [reply tools/iceoryx_build_test.sh:267]: No it fails, if the variable $BUILD_DIR is not empty , but we set it always in this this script independent if the build directory exist or not.
I've added the `no-build` parameter to this script as you suggested here: https://github.com/eclipse-iceoryx/iceoryx/issues/920

File: CONTRIBUTING.md (hunk 2 above)
Comment (dkroenke) [reply CONTRIBUTING.md]: Added

File: tools/scripts/iceoryx_release.sh (hunk 5 above)
Comment (dkroenke) [reply tools/scripts/iceoryx_release.sh:46]: Not sure, the linter proposed me to add some extra double-quotes

File: tools/iceoryx_build_test.sh (hunk 3 above)
Comment (elBoberido) [reply tools/iceoryx_build_test.sh:267]: Please also link to #920 in the PR description

File: tools/iceoryx_build_test.sh [hunk 6]
```diff
@@ -353,13 +361,13 @@ for COMPONENT in $COMPONENTS; do
     case "$1" in
         "unit")
             if [ ! -f testresults/"$COMPONENT"_ModuleTestResults.xml ]; then
-                echo "xml:""$COMPONENT""_ModuletestTestResults.xml not found!"
+                echo "xml: ${COMPONENT}_ModuletestTestResults.xml not found!"
```
Comment (elBoberido) [inline tools/iceoryx_build_test.sh:364]: Now there is an extra space which wasn't there. Is this correct and/or necessary?

Comment (dkroenke) [reply tools/iceoryx_build_test.sh]: Removed it

**Review by mossmaurice — APPROVED** [review]
LGTM!

**Pull request merged**